tests/e2e_tests.c

```diff
@@ -49,48 +53,82 @@ static void	finally(int *out, int *save)
 	close(*save);
 }
 
+#ifdef FT_CALLOC_IF_TRAPPED
 static void	finally_err(int *out, int *save)
 {
 	fflush(stderr);
 	close(*out);
 	dup2(*save, fileno(stderr));
 	close(*save);
 }
+#endif
 
 static int	file_compare(char *exp_contens, char *act_fname)
 {
-	int fc = open(act_fname, O_RDONLY, 0600);
-	int len_exp = ft_strlen(exp_contens);
-	char *act = ft_calloc(sizeof(char), len_exp * 2 + 1);
+	#ifdef DEBUG
+	fprintf(stderr, "comparing files\n");
+	fflush(stderr);
+	fprintf(stderr, "expected [%p]\n", exp_contens);
+	fflush(stderr);
+	fprintf(stderr, "expected [%s]\n", exp_contens);
+	fflush(stderr);
+	fprintf(stderr, "actual fname [%s]\n", act_fname);
+	fflush(stderr);
+	#endif
+	FILE *f = fopen(act_fname, "rb");
+	fseek(f, 0, SEEK_END);
+	long fsize = ftell(f);
+	fseek(f, 0, SEEK_SET);
+	char *act = ft_calloc(sizeof(char), fsize + 1);
 	assert(act);
-	int len_act = read(fc, act, len_exp * 2);
-	assert(len_act - len_exp <= 3); // expected result doesn't store a number
-	int comp_res = ft_strncmp(exp_contens, act, ft_strlen(exp_contens) - 1);
+	fread(act, fsize, 1, f);
+	act[fsize] = 0;
+	fclose(f);
+
+	regex_t re;
+	char	*exp_re = ft_strjoin(exp_contens, "(Total ft_calloc_calls: [0-9]*\n)?");
+	assert(exp_re);
+	assert(regcomp(&re, exp_re, REG_EXTENDED | REG_NOSUB) == 0);
+	int comp_res = regexec(&re, act, 0, NULL, 0);
 	#ifdef DEBUG
-	ft_printf("comparison result %i, expected:[%s] (%i symbols)\nactual:[%s] (%i symbols)\n", comp_res, exp_contens, len_exp, act, len_act);
+	int	diff = -1;
+	for (int i = 0; i < fsize; i ++)
+		if (act[i] != exp_re[i])
+		{
+			diff = i;
+			break ;
+		}
```

Comment:
ok, in output in test there is a messages indicated dismatch of char *act and char *exp_re. for example:
```
"strncmp -84
start to differ from index 106"
```

what does it meaning, the error?

Reply:
This is strcmp error. This is possible, since the actual result should be matched to regexp, not strcompared to it. Regexps contain special characters in most cases, hence direct comparison will fail. Hence strcmp result is not asserted to be 0. This output is for my reference.